File: C++/프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(string X, string Y) {
    string answer = "";
    vector<int> xCnt(10), yCnt(10);
    
    for(int i = 0; i < max(X.size(), Y.size()); i++) {
        if(i < X.size()) xCnt[X[i] - '0']++;
        if(i < Y.size()) yCnt[Y[i] - '0']++;
    }
    
    for(int i = 9; i >= 0; i--) {
        for(int j = 0; j < min(xCnt[i], yCnt[i]); j++) {
            answer += to_string(i);
        }
    }
    
    if(answer.empty()) return "-1";
    else if (answer[0] == '0') return "0";
    return answer;
}

File: Python3/프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.py
def solution(X, Y):
    answer = ''
    X_cnt = [X.count(str(i)) for i in range(10)]
    Y_cnt = [Y.count(str(i)) for i in range(10)]
    
    for i in range(10):
        answer = str(i) * min(X_cnt[i],Y_cnt[i]) + answer
    
    if not answer:
        return "-1"
    if answer[0] == '0':
        return "0"
    return answer